nixos/modules/services/home-automation/zwave-js-ui.nix

```diff
@@ -0,0 +1,84 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    mkIf
+    mkEnableOption
+    mkOption
+    mkPackageOption
+    types
+    ;
+  cfg = config.services.zwave-js-ui;
+in
+{
+  options.services.zwave-js-ui = {
+    enable = mkEnableOption "zwave-js-ui";
+
+    package = mkPackageOption pkgs "zwave-js-ui" { };
+
+    serialPort = mkOption {
+      type = types.path;
+      description = ''
+        Serial port for the Z-Wave controller.
+
+        Only used to grant permissions to the device; must be additionally configured in the application
+      '';
+      example = "/dev/serial/by-id/usb-example";
+    };
+  };
+  config = mkIf cfg.enable {
+    systemd.services.zwave-js-ui = {
+      environment = {
+        STORE_DIR = "/var/lib/zwave-js-ui";
+        ZWAVEJS_EXTERNAL_CONFIG = "/var/lib/zwave-js-ui/.config-db";
+      };
```

Comment:
Add support for passing environment variables. This could be `services.zwave-js-ui.settings`, and example options could be host and port. The `settings` option should reference this page:

https://zwave-js.github.io/zwave-js-ui/#/guide/env-vars